### ISSUE TITLE
Wierd bug: can't read certain PDB entries

### ISSUE THREAD

**Issue by lyu18** [Acellera/htmd#725]
Hi,

I use this to create new `Molecule` objects:

```python
from htmd.molecule.molecule import Molecule
mol = Molecule('filename')
```

Per documentation, `Molecule` module first assumes `filename` is a path to a local file and try to open the file. When this fails it assumes it is a PDB ID and fetch it from the PDB server. This works in 99% of the cases, but not for certain PDB entries. Two such instances I found are `2pgz` and `1ogz`:

```python
mol = Molecule('2pgz')
```

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-2-3bb9e7612235> in <module>()
----> 1 mol = Molecule('2pgz')

~/miniconda/lib/python3.6/site-packages/htmd/molecule/molecule.py in __init__(self, filename, name, **kwargs)
    195 
    196         if filename is not None:
--> 197             self.read(filename, **kwargs)
    198 
    199     @staticmethod

~/miniconda/lib/python3.6/site-packages/htmd/molecule/molecule.py in read(self, filename, type, skip, frames, append, overwrite, keepaltloc, guess, guessNE, _logger)
    869 
    870         for fname, frame in zip(filename, frames):
--> 871             fname = self._unzip(fname)
    872             ext = self._getExt(fname, type)
    873 

~/miniconda/lib/python3.6/site-packages/htmd/molecule/molecule.py in _unzip(self, fname)
    967             import gzip
    968             from htmd.util import tempname
--> 969             with gzip.open(fname, 'r') as f:
    970                 fname = tempname(suffix='.{}'.format(fname.split('.')[-2]))
    971                 with open(fname, 'w') as fo:

~/miniconda/lib/python3.6/gzip.py in open(filename, mode, compresslevel, encoding, errors, newline)
     51     gz_mode = mode.replace("t", "")
     52     if isinstance(filename, (str, bytes, os.PathLike)):
---> 53         binary_file = GzipFile(filename, gz_mode, compresslevel)
     54     elif hasattr(filename, "read") or hasattr(filename, "write"):
     55         binary_file = GzipFile(None, gz_mode, compresslevel, filename)

~/miniconda/lib/python3.6/gzip.py in __init__(self, filename, mode, compresslevel, fileobj, mtime)
    161             mode += 'b'
    162         if fileobj is None:
--> 163             fileobj = self.myfileobj = builtins.open(filename, mode or 'rb')
    164         if filename is None:
    165             filename = getattr(fileobj, 'name', '')

FileNotFoundError: [Errno 2] No such file or directory: '2pgz'
```

Looks like it still thinks `2pgz` is a local file and fails to fetch it from PDB. Of course, both `2pgz` and `1ogz` are legit PDB entries.

When use all uppercase, i.e. `2PGZ` and `1OGZ` work as intended. But I feel it's more common in cheminformatics to use lowercase PDB IDs.

**Comment by j3mdamas** [comment]
maybe it's thinking it's a gzip file? I'll have to check

**Comment by j3mdamas** [comment]
it is, it's in the trace. Let's see how to fix that

**Comment by stefdoerr** [comment]
ahah yes, it assumes it's a gz. I will fix the code for that check. It should check for `.gz` and not `gz`. Sorry for the stupid bug

**Comment by stefdoerr** [comment]
https://github.com/Acellera/htmd/pull/726 this fixes it

**Comment by lyu18** [comment]
Wow that is fast. Thanks :)

**Comment by j3mdamas** [comment]
@lyu18 you can use the master repository as soon as it's merged, or wait for version 1.13.6

**Comment by stefdoerr** [comment]
I'll still wait for the tests to pass so that Joao doesn't murder me :) so in 45min more or less we can merge
